Flashcards.kronyer/Flashcards/DataAccess.cs

```diff
@@ -0,0 +1,322 @@
+            return new List<Flashcard>();
+        }
+    }
+
+    internal void InsertFlashcard(Flashcard flashcards)
+    {
+        try
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string inserQuery = @"
+            INSERT INTO Flashcards (Question, Answer, StackId) VALUES (@Question, @Answer, @StackId)";
+
+                connection.Execute(inserQuery, new { flashcards.Question, flashcards.Answer, flashcards.StackId });
+            }
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e.Message);
+        }
+    }
+
+
+
+    internal void UpdateFlashcard(int flashcardId, Dictionary<string, object> propertiesToUpdate)
+    {
+        using (var connection = new SqlConnection(ConnectionString))
+        {
+            connection.Open();
+
+            string updateQuery = "UPDATE flashcards SET ";
+            var parameters = new DynamicParameters();
+
+            foreach (var kvp in propertiesToUpdate)
+            {
+                updateQuery += $"{kvp.Key} = @{kvp.Key}, ";
+                parameters.Add(kvp.Key, kvp.Value);
+            }
+
+            updateQuery = updateQuery.TrimEnd(',', ' ');
+
+            updateQuery += " WHERE Id = @Id;";
+            parameters.Add("Id", flashcardId);
+
+            connection.Execute(updateQuery, parameters);
+        }
+    }
+
+    internal void DeleteTables()
+    {
+        try
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+
+                string dropFlashcardsTableSql = @"DROP TABLE Flashcards";
+                connection.Execute(dropFlashcardsTableSql);
+
+                string dropStacksTableSql = @"DROP TABLE Stacks";
+                connection.Execute(dropStacksTableSql);
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
+    internal void DeleteFlashcard(int id)
+    {
+        try
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string deleteQuery = "DELETE FROM flashcards WHERE Id = @Id";
+
+                connection.Execute(deleteQuery, new { Id = id });
+
+
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
+    internal void DeleteStack(int id)
+    {
+        try
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string deleteQuery = "DELETE FROM stacks WHERE Id = @Id";
+                connection.Execute(deleteQuery, new { Id = id });
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
```

Comment:
Don't forget to tidy up extra spaces 😅 check other classes and clean them up too 🚀